### PR TITLE
Bug fixes

### LINKED ISSUE #162: [PE-D][Tester B] Cannot `edit` room numbers of a person to another person’s room number even though they're on different dates
Here I want to change the room number of guest 1 to the same room number of guest 2, which should have been possible because they are occupying it on different dates.

However after `edit 1 rm/06-84`

Perhaps including a date check might solve this bug?

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: NicholasTYD/ped#3

### LINKED ISSUE #168: [PE-D][Tester D] Lack of details for download instructions.
Only the page for Mac OS is shown, and not windows. However, the images of the app itself is shown in Windows OS. Maybe consistency will help or adding of both OS.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: mohamedsaf1/ped#5

### LINKED ISSUE #182: [PE-D][Tester C] Missing spacing in the message constraint of IsRoomClean field
-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: ReaganTan00/ped#3

### LINKED ISSUE #186: [PE-D][Tester E] Program does not open normally on Linux
### Description:

Program fails to open at all on Ubuntu 22.04.1 Linux machines with the command

```
java -jar guestbook.jar
```

### Steps to reproduce:
1. Use an Ubuntu machine with 22.04.1

2. Download the jar file and open in terminal with `java -jar guestbook`

3. Observe the critical error message shown

4. Double clicking will not open the file either.

5. The program can be executed and run normally with additional parameters `java -Djdk.gtk.version=2 -jar guestbook.jar`

### Suggestions:

This is caused by an incompatibility with JavaFX and Wayland on Ubuntu. Updating the JavaFX version in dependencies to version 11.0 to 11.0.2 in the `build.gradle` file can fix this.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: rui-han-crh/ped#10

### PR DESCRIPTION
**Pull request by jehousoh**
- Update Use Cases in DG
- Update JavaFX version from 11.0 to 11.0.2 to fix failure to launch on linux issues
- Update GuestBookIcon.png to Windows version
- Update room occupied message
- Update sample data to all be in similar dateRanges

Closes #186, Closes #182, Closes #168, Closes #162

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-    Use case resumes at step 2.
+  
+    Use case ends.
 
 **Use case 2: Search for a guest**
 
 **MSS**
 
-1. User requests to search for a guest.
+1. User requests to search for a guest through any of the guest's details.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "User requests to search for a guest by specifying keywords from any of the target guest's details"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 2. System returns the guest.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "System returns a list of matching guests including the target guest"

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
    Use case ends.
 
 **Extensions**
-* 1a. The guest is does not exist in the system.
+* 1a. The guest's details does not exist in the system.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The target guest's details do not contain matching keywords with the query"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-  Use case ends.
+    * 1a1. System returns no guests.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "System does not return the target guest in the results"

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 1. User <u>searches for guest (Use case 2)</u>.
 2. User requests to edit a guest's details.
-3. System updates the guest.
+3. System updates the guest's details.
 
    Use case ends.
 
 **Extensions**
-* 2a. The edit data is invalid.
+* 2a. The edit name already exists.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The new name provided already belongs to another existing guest"

Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The new name provided already belongs to another existing guest"

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
     * 2a1. System shows an error message.
 
-      Use case resumes at step 2.
+      Use case ends.
 
-**Use case 4: Delete a guest**
+* 2b. The edit room is already occupied by another guest.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The new room provided has already been assigned to another guest"

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 2b1. System shows an error message.
 
-1.  User requests to list guests.
-2.  System shows a list of guests.
-3.  User requests to delete a specific guest in the list.
-4.  System deletes the guest.
+      Use case ends.
 
-    Use case ends.
+* 2c. The edit data is invalid.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The new data provided is invalid"

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -590,86 +590,92 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
   Use case ends.
 
+* 2a. The given index is invalid.
+
+    * 2a1. System shows an error message.
 
-**Use case 6: Mark room clean status to "no" of all guests**
+      Use case ends.
+
+**Use case 6: Mark all guests' room clean status to "no"**
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we use the code block `no`? (Applies to other occurrences in the document)

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -689,18 +695,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
    Use case ends.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md:695]: For use case 7, can we say "modify a guest's bill"? Or "Add to or deduct from a guest's bill"?

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -689,18 +695,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
 **Extensions**
-* 2a. The update data is invalid.
+* 2a. The guest's bill is negative.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The guest's updated bill is negative"

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -689,18 +695,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
     * 2a1. System shows an error message.
 
-      Use case resumes at step 2.
+      Use case ends.
+
+* 2b. The update data is invalid.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we say "The value provided is invalid"?

File: src/main/java/seedu/guest/logic/commands/AddCommand.java [hunk 12]
```diff
@@ -40,7 +40,7 @@ public class AddCommand extends Command {
 
     public static final String MESSAGE_SUCCESS = "New guest added: %1$s";
     public static final String MESSAGE_DUPLICATE_GUEST = "This guest already exists in the guest book";
-    public static final String MESSAGE_ROOM_OCCUPIED = "The room is currently occupied by a guest";
+    public static final String MESSAGE_ROOM_OCCUPIED = "All guests are required to have unique rooms";
```
Comment (JustinPeng13) [inline src/main/java/seedu/guest/logic/commands/AddCommand.java]: Can we say "Each guest is required to have a unique room"?

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]
Just some language comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed.

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed.

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed.

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed.

File: src/main/java/seedu/guest/logic/commands/AddCommand.java (hunk 12 above)
Comment (jehousoh) [reply src/main/java/seedu/guest/logic/commands/AddCommand.java]: Agreed.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed!

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md:695]: Agreed!

File: docs/DeveloperGuide.md (hunk 10 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed!

File: docs/DeveloperGuide.md (hunk 11 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Agreed!

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -341,25 +341,25 @@ The following activity diagram summarizes what happens when a user enters an `ed
...
-* When the user executes the `markroomsunclean` command, GuestBook will retrieve the list of all the guests that requires editing of the isRoomClean status.
-* A new list of guests will be created with isRoomClean fields set to "no". The other details of the guests will remain unchanged.
+* When the user executes the `markroomsunclean` command, GuestBook will retrieve the list of all the guests that requires editing of the `isRoomClean` status.
+* A new list of guests will be created with `isRoomClean` fields set to `no`. The other details of the guests will remain unchanged.
 * Finally, the new list of guests is set to the model.
 
 <span style="font-size: large; color: #e46c0a">Design Considerations:</span>
-**Aspect: The scope at which the command changes all guests' isRoomClean statuses**
+**Aspect: The scope at which the command changes all guests' `isRoomClean` statuses**
 * Alternative 1: Allow `markroomsunclean` command to operate only on the last shown list instead of the entire list. This is to standardise how edits are made across the commands (e.g. edit and delete).
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we use code blocks for `edit` and `delete`?

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -341,25 +341,25 @@ The following activity diagram summarizes what happens when a user enters an `ed
...
   * Pros: This might be more intuitive for users, as `edit` and `delete` commands work only on the last shown lists.
-  * Cons: User is unable to change all the guests' isRoomClean statuses in a single command.
-* Alternative 2 (current choice): Allow `markroomsunclean` command to change all guests' isRoomClean statuses in GuestBook instead of the last shown list.
-  * Pros: User is able to change all the guests' isRoomClean statuses in a single command.
+  * Cons: User is unable to change all the guests' `isRoomClean` statuses in a single command.
+* Alternative 2 (current choice): Allow `markroomsunclean` command to change all guests' `isRoomClean` statuses in GuestBook instead of the last shown list.
+  * Pros: User is able to change all the guests' `isRoomClean` statuses in a single command.
   * Cons: There is less flexibility in marking groups of guests' room as unclean.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should this be "rooms" not "room"?

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -483,18 +483,18 @@ As the users of this application would be hotel mangers of small hotels, there w
...
+| **Clear**              | `clear`                                                                                                                   | Clears all guest entries in GuestBook.                                                                                                                  |                                                                                                                                                      |
+| **Delete**             | `delete INDEX`                                                                                                            | `delete 3`<br> Deletes the third guest in the current list.                                                                                             |
+| **Edit**               | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [rm/ROOM] [dr/DATE_RANGE] [ng/NUMBER_OF_GUESTS] [rc/IS_ROOM_CLEAN] [rq/REQUEST]` | `edit 2 rc/yes` <br> Edits the second guest in the current list by changing the guest's the room clean status to "yes".                                 |
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Is there an extra "the" before "room clean"? And can we use code block for `yes`?

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -483,18 +483,18 @@ As the users of this application would be hotel mangers of small hotels, there w
...
+| **Exit**               | `exit`                                                                                                                    | Exits and closes the GuestBook application.                                                                                                             |
+| **Find**               | `find KEYWORD [MORE_KEYWORDS]`                                                                                            | `find James Jake` <br> Searches the entire GuestBook for fields matching "James" and "Jake" and returns the matching guests.                            |
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Should we use code blocks for `James` and `Jake`? What do you think?

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (JustinPeng13) [reply docs/DeveloperGuide.md:695]: I think there are still similar changes to be made within the use case steps!

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]
Found a few more nits to fix!

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md:695]: Added!

File: docs/DeveloperGuide.md (hunk 13 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Added!

File: docs/DeveloperGuide.md (hunk 14 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Yes. Changed!

File: docs/DeveloperGuide.md (hunk 15 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Yes changed!

File: docs/DeveloperGuide.md (hunk 16 above)
Comment (jehousoh) [reply docs/DeveloperGuide.md]: Changed.

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -689,8 +689,8 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **MSS**
 
 1. User <u>searches for guest (Use case 2)</u>.
-2. User requests to add or deduct to a guest's bill.
-3. System add or deduct to original guest's bill.
+2. User requests to add to or deduct from a guest's bill.
+3. System adds to or deducts from  the original guest's bill.
```
Comment (JustinPeng13) [inline docs/DeveloperGuide.md]: Is there an extra space here?

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]

**Review by JustinPeng13 — APPROVED** [review]
lgtm

**Pull request merged**